SecretSanta/src/SecretSanta.Api/Controllers/GroupController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SecretSanta.Business;
+using SecretSanta.Data;
+
+namespace SecretSanta.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class GroupController : ControllerBase
+    {
+        private IGroupService GroupService { get; }
+
+        public GroupController(IGroupService groupService)
+        {
+            GroupService = groupService ?? throw new System.ArgumentNullException(nameof(groupService));
+        }
+
+        // GET: https://localhost/api/User
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<List<Group>>> Get()
+        {
+            if (await GroupService.FetchAllAsync() is List<Group> groups && groups != null && groups.Count != 0)
+            {
+                return Ok(groups);
+            }
+            return NotFound();
+        }
+
+        // GET: api/Author/5
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<Group>> Get(int id)
+        {
+            if (await GroupService.FetchByIdAsync(id) is Group group && group != null)
```

Comment:
The null check here is not needed. The pattern matching ensures that the value is not null.
```suggestion
            if (await GroupService.FetchByIdAsync(id) is Group group)
```